crates/rust-analyzer/src/main_loop.rs

```diff
@@ -146,11 +158,13 @@ impl GlobalState {
             self.register_did_save_capability();
         }
 
-        self.fetch_workspaces_queue.request_op("startup".to_owned(), false);
-        if let Some((cause, force_crate_graph_reload)) =
-            self.fetch_workspaces_queue.should_start_op()
-        {
-            self.fetch_workspaces(cause, force_crate_graph_reload);
+        if self.config.discover_command().is_none() {
```

Comment:
this feels a little gross, but I can't think of a better way to do this

Reply:
this is still gross, and i'm struggling to make this better with the current architecture with OpQueues—the state machine is pretty defuse across all of the rust-analyzer crate. I'll explain the pitfalls of the approach I took in a larger comment.